api/v1alpha1/statuscheck_types.go

```diff
@@ -0,0 +1,230 @@
+
+	// Duration defines the duration of the whole status check if the
+	// number of failed execution does not exceed the failure threshold.
+	// Duration is available to both `Synchronous` and `Continuous` mode.
+	// A duration string is a possibly signed sequence of
+	// decimal numbers, each with optional fraction and a unit suffix,
+	// such as "300ms", "-1.5h" or "2h45m".
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +optional
+	Duration *string `json:"duration,omitempty" webhook:"Duration"`
+
+	// TimeoutSeconds defines the number of seconds after which
+	// an execution of status check times out.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
+
+	// IntervalSeconds defines how often (in seconds) to perform
+	// an execution of status check.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	IntervalSeconds int `json:"intervalSeconds,omitempty"`
+
+	// FailureThreshold defines when an execution of status check
+	// fails, it will try FailureThreshold times before giving up.
+	// +optional
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Minimum=1
+	FailureThreshold int `json:"failureThreshold,omitempty"`
+
+	// SuccessThreshold defines the minimum consecutive successes
+	// for the status check to be considered successful.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	SuccessThreshold int `json:"successThreshold,omitempty"`
+
+	// RecordsHistoryLimit defines the number of record to retain.
+	// +optional
+	// +kubebuilder:default=100
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	RecordsHistoryLimit int `json:"recordsHistoryLimit,omitempty"`
+
+	// +optional
+	*EmbedStatusCheck `json:",inline,omitempty"`
+}
+
+type StatusCheckStatus struct {
+	// StartTime represents time when the status check started to execute.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime represents time when the status check was completed.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Count represents the total number of the status check executed.
+	// +optional
+	Count int64 `json:"count,omitempty"`
+
+	// Conditions represents the latest available observations of a StatusCheck's current state.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []StatusCheckCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Records contains the history of the execution of StatusCheck.
+	// +optional
+	Records []StatusCheckRecord `json:"records,omitempty"`
+}
+
+type StatusCheckOutcome string
+
+const (
+	StatusCheckOutcomeSuccess StatusCheckOutcome = "Success"
+	StatusCheckOutcomeFailure StatusCheckOutcome = "Failure"
+	StatusCheckOutcomeUnknown StatusCheckOutcome = "Unknown"
```

Comment:
Which situation would let it become `Unknown`?

Reply:
I think it's actually kind of useless, updated!